kernel/mutex.c

```diff
@@ -276,6 +295,13 @@ int z_impl_k_mutex_unlock(struct k_mutex *mutex)
 
 
 k_mutex_unlock_return:
+#else
+	ARG_UNUSED(mutex);
+
+	__ASSERT(!arch_is_in_isr(), "mutexes cannot be used inside ISRs");
+
+	SYS_PORT_TRACING_OBJ_FUNC_ENTER(k_mutex, unlock, mutex);
```

Comment:
This part is dulicated in both parts of the`#ifdef`. It can probably just be refactored to the top of the function, outside the `#ifdef`, and then the `#else` can be removed.

Reply:
See fixup commit. It is not clear to me if I should keep `ARG_UNUSED(...)` because warning as error could let to fail. Should I keep it on the `#else` maybe?

Reply:
`ARG_UNUSED(foo)` can safely be added even when `foo` is used. Looks like the particular warning is already resolved.

Reply:
Sorry for the delay - it's been an extremely intense past month (6 months 😅)